new-lamassu-admin/src/pages/Maintenance/MachineDetailsCard.js

```diff
@@ -69,22 +79,82 @@ const Item = ({ children, ...props }) => (
 )
 
 const MachineDetailsRow = ({ it: machine, onActionSuccess }) => {
-  const [action, setAction] = useState(null)
+  const [action, setAction] = useState({ command: null })
   const [errorMessage, setErrorMessage] = useState(null)
+  const [machineState, setMachineState] = useState(null)
   const classes = useMDStyles()
 
+  const [fetchMachineEvents, { loading: loadingEvents }] = useLazyQuery(
+    MACHINE,
+    {
+      variables: {
+        deviceId: machine.deviceId
+      },
+      onCompleted: machineEventsLazy => {
+        if (machineEventsLazy.machine.latestEvent) {
+          setMachineState(
+            JSON.parse(machineEventsLazy.machine.latestEvent.note).state
+          )
+        }
+      }
+    }
+  )
+
+  useEffect(() => {
+    if (action.command === 'restartServices') {
+      fetchMachineEvents()
+    }
+  }, [action.command, fetchMachineEvents])
+
+  // on slow connections, we could have the confirm dialog open before the query is fetched from the server
+  // this would make the warning not appear in some situations
+  // so, have to check if we should update the message on completion of the fetch query
```

Comment:
Is this comment still relevant? Didn't we add the behavior to make the button disabled?